lib/galaxy/webapps/galaxy/api/workflows.py

```diff
@@ -117,33 +117,41 @@ def get_workflows_list( self, trans, for_menu, kwd ):
         missing_tools = util.string_as_bool( kwd.get( 'missing_tools', 'False' ) )
         rval = []
         filter1 = ( trans.app.model.StoredWorkflow.user == trans.user )
+        user = trans.get_user()
         if show_published:
             filter1 = or_( filter1, ( trans.app.model.StoredWorkflow.published == true() ) )
         for wf in trans.sa_session.query( trans.app.model.StoredWorkflow ).filter(
                 filter1, trans.app.model.StoredWorkflow.table.c.deleted == false() ).order_by(
                 desc( trans.app.model.StoredWorkflow.table.c.update_time ) ).all():
-            if for_menu:
-                item = wf.to_dict()
-            else:
-                item = wf.to_dict( value_mapper={ 'id': trans.security.encode_id } )
-                encoded_id = trans.security.encode_id(wf.id)
-                item['url'] = url_for('workflow', id=encoded_id)
+
+            item = wf.to_dict( value_mapper={ 'id': trans.security.encode_id } )
+            encoded_id = trans.security.encode_id(wf.id)
+            item['url'] = url_for('workflow', id=encoded_id)
             item['owner'] = wf.user.username
             item['number_of_steps'] = len( wf.latest_workflow.steps )
+            item['show_in_tool_panel'] = False
+            item['wf_id'] = wf.id
```

Comment:
@mvdbeek @guerler Thanks for the review!
I removed the unencoded workflow id from the api and also removed `get_workflow_menu` api call which fetched workflow ids present in the tools panel. Now, there is a flag in the workflows set itself (boolean - `show_in_tool_panel`) which determines whether any workflow is to be shown in the tools panel. Now, we just use encoded workflow id.